File: LogicaAlgoritmosEmC/PesquisaBinaria.c
/*This space is intended for testing algorithms using binary search.
And all exercises related to binary search will be tested here.*/ 

#include <stdio.h>
int ListaBinaria (int list[], int lenth, int number);


int main(){
    int list[8]={1,3,5,7,9,11,13,17};
    int number;
    printf("type a number: ");
    scanf("%d", &number);
    printf("you typed %d\n", number);
   
 
       

 

    int result = ListaBinaria(list, 8, number);
    if (result != -1) { 
        printf("\nThe number %d is at position %d\n", number, result);
    }
    else 
        printf("\nThe number %d was not found\n", number);

    return 0; 

    
}

int ListaBinaria (int list[], int lenth, int number) {
    int low = 0;
    int high = lenth - 1;
    while (low <= high) {
        int mid = (low + high) / 2;
        int chute = list[mid];
        if (chute == number) {
            return mid;
        }
        if (chute > number) {
            high = mid - 1;
        } else {
            low = mid + 1;
        }
    }
    return -1;

}

File: LogicaAlgortimosEmPython/PesquisaBinaria.py
#Here is a small demonstration of the binary search code in Python.

number=int(input("type a number: "))

alvo=number
def pesquisa_binaria(list, alvo):
   
    low= 0 
    high = len(list) - 1 
   
    while low <= high: 
        middle = (low + high) // 2 
        chute = list[middle]
        if chute == alvo:   
            return middle
        if chute > alvo: 
            high = middle - 1
        else:
            low = middle + 1
    return None
minha_lista = [1, 3, 5, 7, 9, 50,55,60]
print("The result is " + str(pesquisa_binaria(minha_lista, number)))
